Added option to plot TPF background to `TargetPixelFile.plot()`

ojhall94 opened the pull request:
For cases where there are strong background features (e.g. near the corners of TESS CCDs) in Target Pixel File data, it might be useful to be able to plot the TPF background flux using the Lightkurve infrastructure.

This PR changes the keyword arguments of `TargetPixelFile.plot()`, so that:

- `bkg=True` plots the TPF flux background _only_
- `add_bkg=True` plots the TPF flux with the background added (this functionality was previously under the `bkg=True` argument).

If `bkg=True` and `add_bkg=True` are passed, the `bkg` keyword is ignored.

Example usage:
![image](https://user-images.githubusercontent.com/22291663/82350150-e1f0a000-99f2-11ea-84a2-c8658d3c56e9.png)

barentsen commented:
(The two remote-data test runs are failing due to no fault of your own.  Please ignore.  I hope it'll get fixed shortly.)

ojhall94 commented:
Great suggestion @barentsen, updated the PR. Added in the unit tests for the new functionality too. It still has an issue--- TESS TPFs store the COSMIC_RAYS data in a different extension of the FITS file. I'm guessing this can be alleviated in a clever way, but can't think of it right now. Will give another pass over with your comments.

barentsen commented:
Thanks @ojhall94.  This looks good!

Just one more request: to keep the function as flexible and future-proof as possible, I'd recommend removing the bit where you check the value of `column` against the hard-coded list of column names, and instead do something like this:

```python
try:
    data_to_plot = self.hdu[1].data[column]
except KeyError:
    raise ValueError(f"column {column} does not exist in this file")
```

This will also resolve the issue with `COSMIC_RAYS`, i.e. it'll work for Kepler and raise a ValueError for TESS, which is fine.

(You can of course keep mentioning the typically available column names in the docstring, that is fine!)

ojhall94 commented:
Great suggestion, made the changes.

barentsen commented on lightkurve/targetpixelfile.py:
```diff
@@ -734,8 +734,11 @@ def plot(self, ax=None, frame=0, cadenceno=None, bkg=False, aperture_mask=None,
-        bkg : bool
+        bkg: bool
             If True, background will be added to the pixel values.
+        column :
```
Make sure we specify the expected data type:
```suggestion
        column : str
```

barentsen commented on lightkurve/targetpixelfile.py:
```diff
@@ -762,22 +765,37 @@ def plot(self, ax=None, frame=0, cadenceno=None, bkg=False, aperture_mask=None,
                 raise ValueError("cadenceno {} is out of bounds, "
                                  "must be in the range {}-{}.".format(
-                                     cadenceno, self.cadenceno[0], self.cadenceno[-1]))
+                                     cadenceno, self.cadenceno[0], self.cadenceno[-1]))      
```
Remove trailing whitespace:
```suggestion
                                     cadenceno, self.cadenceno[0], self.cadenceno[-1]))
```

barentsen commented on lightkurve/targetpixelfile.py:
```diff
@@ -762,22 +765,37 @@ def plot(self, ax=None, frame=0, cadenceno=None, bkg=False, aperture_mask=None,
-            ax = plot_image(pflux, ax=ax, title=title, extent=img_extent,
-                            show_colorbar=show_colorbar, **kwargs)
+            ax = plot_image(data_to_plot, ax=ax, title=title, extent=img_extent,
+                            show_colorbar=show_colorbar, clabel = clabels[column], **kwargs)
```
Calling `clabel = clabels[column]` would crash if someone did call this function on a fancy TargetPixelFile that has an extra column we didn't expect.  A good solution here is to use `dict.get(key, default)`, which returns `default` if `key` is not available in the dictionary:

```suggestion
                            show_colorbar=show_colorbar, clabel = clabels.get(column, column), **kwargs)
```

barentsen commented on lightkurve/targetpixelfile.py:
```diff
@@ -762,22 +765,37 @@ def plot(self, ax=None, frame=0, cadenceno=None, bkg=False, aperture_mask=None,
+                data_to_plot = self.hdu[1].data[column][self.quality_mask][frame]
+        except KeyError:
+            raise ValueError("column must be one of the following: ('FLUX','FLUX_ERR',"
+                                "'FLUX_BKG','FLUX_BKG_ERR','COSMIC_RAYS','RAW_CNTS')")
```
PEP8 recommends aligning the string with the opening delimiter for easier readability:
```suggestion
                             "'FLUX_BKG','FLUX_BKG_ERR','COSMIC_RAYS','RAW_CNTS')")
```

barentsen commented on lightkurve/targetpixelfile.py:
```diff
@@ -734,8 +734,11 @@ def plot(self, ax=None, frame=0, cadenceno=None, bkg=False, aperture_mask=None,
             This argument has priority over frame number.
-        bkg : bool
+        bkg: bool
             If True, background will be added to the pixel values.
```
Let's be slightly more specific:
```suggestion
            If True and `column="FLUX"`, background will be added to the pixel values.
```

ojhall94 commented:
Thanks for checking my code! :) Made the suggested fixes.
